src/core_atmosphere/physics/mpas_atmphys_driver_cloudiness.F

```diff
@@ -106,15 +109,21 @@ subroutine cloudiness_from_MPAS(configs,mesh,diag_physics,sfc_input,its,ite)
  real(kind=RKIND),pointer:: len_disp
  real(kind=RKIND),dimension(:),pointer:: areaCell,meshDensity
  real(kind=RKIND),dimension(:),pointer:: xland
-
+ type(Field1DReal), pointer :: areaCellField
+ type(mpas_coupler_type), pointer :: coupler
 !-----------------------------------------------------------------------------------------------------------------
 
  call mpas_pool_get_config(configs,'config_len_disp',len_disp)
 
- call mpas_pool_get_array(mesh,'areaCell'   ,areaCell   )
- call mpas_pool_get_array(mesh,'meshDensity',meshDensity)
+ call mpas_pool_get_array_gpu(mesh,'areaCell'   ,areaCell   )
+ call mpas_pool_get_field(mesh,'areaCell'   ,areaCellField   )
+ call mpas_pool_get_array_gpu(mesh,'meshDensity',meshDensity)
 
- call mpas_pool_get_array(sfc_input,'xland',xland)
+ call mpas_pool_get_array_gpu(sfc_input,'xland',xland)
+ coupler => areaCellField%block%domain%mpas_cpl
+ if (coupler%role_includes(ROLE_INTEGRATE)) then
+        !$acc update host(areaCell, meshDensity, xland)
```

Comment:
Why do we need to update `areaCell`, `meshDensity`, and `xland` on the host in every call to `cloudiness_from_MPAS`?

Reply:
The update to the host is necessary as the next steps involve CPU code. As we are not tracking which individual variable is modified on GPU in each timestep, we try to update all the variables on host before the start of the CPU code. 
Finally, these update directives will be removed once the entire cloudiness scheme is ported onto GPU.

Reply:
The `areaCell` and `meshDensity` fields don't change in time, and since these are first read from the model ICs by code executing on the CPU (if I understand the program flow correctly), I was just wondering why these updated were needed at all. Did someone find that these were required for correct results at some point in the past? If the plan is to later remove unnecessary transfers, then I won't object to adding these temporarily.

Part of the reason why I've been insisting on commit messages that actually describe what the code changes are doing and why is so that others can understand issues like the one discussed here.

Reply:
We did not find that these were required for correct results. During MPAS 6.x Physics porting, we were not sure about which variable are updated on CPU/GPU, and keeping track of them during the porting would have taken more time. Moreover as these update directives will be removed once the entire scheme is ported, we updated all the variables for time-being.
Except in radiation related subroutines, all the update host in other schemes will be removed entirely. And for radiation, I believe we can later track and remove the unnecessary variables.

Reply:
@mgduda: Yes, these will be removed once the porting has been completed. It is just a precautionary step from our side to avoid any unanticipated bugs into the code. So, in the end, you will not see such updates in the stable versions of openacc code as seen in 6.x. These updates, though unnecessary, will be useful when we hit a bug. It allows to easily track down a variable that is deviating in results. So, would it be ok to keep this temporarily?

Reply:
@raghuraj19 and @Pranay-Reddy-Kommera That makes sense -- thanks very much for the explanation!